From 17cef81b9c03e737f9a65a604e62120ebe0f6bc6 Mon Sep 17 00:00:00 2001
From: Christian Kellner <christian@kellner.me>
Date: Thu, 28 Jun 2018 18:47:07 +0300
Subject: [PATCH] data: make service dbus-activatable

Always install the dbus service file and specify the systemd
service in it should make the service dbus-activatable and
thus we don't need to explicitly enable it (also we have one
less daemon running, if it is not needed).
---
 data/com.feralinteractive.GameMode.service.in |  1 +
 data/meson.build                              | 17 +++++++++--------
 meson.build                                   | 15 +++++++--------
 3 files changed, 17 insertions(+), 16 deletions(-)

diff --git a/data/com.feralinteractive.GameMode.service.in b/data/com.feralinteractive.GameMode.service.in
index 95a594e..5026ee5 100644
--- a/data/com.feralinteractive.GameMode.service.in
+++ b/data/com.feralinteractive.GameMode.service.in
@@ -1,3 +1,4 @@
 [D-BUS Service]
 Name=com.feralinteractive.GameMode
 Exec=@BINDIR@/gamemoded -d
+SystemdService=gamemoded.service
\ No newline at end of file
diff --git a/data/meson.build b/data/meson.build
index 93c03b8..1fafcdf 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -10,16 +10,17 @@ if with_systemd == true
         configuration: data_conf,
         install_dir: path_systemd_unit_dir,
     )
-else
-    # Install the D-BUS service file
-    configure_file(
-        input: 'com.feralinteractive.GameMode.service.in',
-        output: 'com.feralinteractive.GameMode.service',
-        configuration: data_conf,
-        install_dir: path_dbus_service_dir,
-    )
 endif
 
+# Install the D-BUS service file
+configure_file(
+  input: 'com.feralinteractive.GameMode.service.in',
+  output: 'com.feralinteractive.GameMode.service',
+  configuration: data_conf,
+  install_dir: path_dbus_service_dir,
+)
+
+
 # Install the Polkit action file in all cases
 configure_file(
     input: 'com.feralinteractive.GameMode.policy.in',
diff --git a/meson.build b/meson.build
index b01a5cf..d878fa0 100644
--- a/meson.build
+++ b/meson.build
@@ -53,12 +53,12 @@ if with_systemd == true
         pkgconfig_systemd = dependency('systemd')
         path_systemd_unit_dir = pkgconfig_systemd.get_pkgconfig_variable('systemduserunitdir')
     endif
-else
-    # Set the dbus path as appropriate.
-    path_dbus_service_dir = get_option('with-dbus-service-dir')
-    if path_dbus_service_dir == ''
-        path_dbus_service_dir = join_paths(path_datadir, 'dbus-1', 'services')
-    endif
+endif
+
+# Set the dbus path as appropriate.
+path_dbus_service_dir = get_option('with-dbus-service-dir')
+if path_dbus_service_dir == ''
+  path_dbus_service_dir = join_paths(path_datadir, 'dbus-1', 'services')
 endif
 
 path_polkit_action_dir = join_paths(path_datadir, 'polkit-1', 'actions')
@@ -114,11 +114,10 @@ if with_systemd == true
 report += [
     '    systemd user unit directory:            @0@'.format(path_systemd_unit_dir),
 ]
-else
+endif
 report += [
     '    D-BUS service directory:                @0@'.format(path_dbus_service_dir),
 ]
-endif
 
 report += [
 
-- 
2.17.1

